lib/write-to-stream.ts

```diff
@@ -4,20 +4,28 @@ import { AnyIterable } from './types'
 interface IWritable {
   once: any
   write: any
+  removeListener: any
 }
 
-function waitForDrain(stream: IWritable) {
+function once(event: string, stream: IWritable): Promise<any> {
   return new Promise(resolve => {
-    stream.once('drain', resolve)
+    stream.once(event, resolve)
   })
 }
 
 async function _writeToStream(stream: IWritable, iterable: AnyIterable<any>) {
+  let errorListener
+  const errorPromise = new Promise((resolve, reject) => {
+    errorListener = reject
+    stream.once('error', reject)
+  })
   for await (const value of iterable) {
     if (stream.write(value) === false) {
-      await waitForDrain(stream)
+      await Promise.race([errorPromise, once('drain', stream)])
     }
   }
+  await Promise.race([errorPromise, Promise.resolve()])
+  stream.removeListener('error', errorListener)
```

Comment:
I think that you might have a race condition here.

I think that it's possible that the `errorListener` could be called after the last `Promise.race` and before it's removed. That would lead you to silently eating that error.

Reply:
I'm currently testing with a simple script piped to `| head -n1` and the error gets silently eaten.

Reply:
I think that the safest thing would be for you to remove the listener first and then to race the two promises. That way, if you have accumulated an error then you'll be end up throwing it and you can't accidentally silently eat up another error.